docs/configuring-a-resource.md

```diff
@@ -0,0 +1,410 @@
+## Configuring a Resource
+
+[Terrajet] generates as much as it could using the available information in the
+Terraform resource schema. This includes an XRM-conformant schema of the
+resource, controller logic, late initialization, sensitive data handling etc.
+However, there are still couple of information that requires some input
+configuration which could easily be provided by checking the Terraform
+documentation of the resource:
+
+- [External name]
+- [Cross Resource Referencing]
+- [Additional Sensitive Fields and Custom Connection Details]
+- [Late Initialization Behavior]
+
+### External Name
+
+Crossplane uses an annotation in managed resource CR to identify the external
+resource which is managed by Crossplane. See [the external name documentation]
+for more details. The format and source of the external name depends on the
+cloud provider; sometimes it could simply be the name of resource
+(e.g. S3 Bucket), and sometimes it is an auto-generated id by cloud API
+(e.g. VPC id ). This is something specific to resource, and we need some input
+configuration for terrajet to appropriately generate a resource.
+
+Since Terraform already needs the same identifier to import a resource, most
+helpful part of resource documentation is the [import section].
+
+This is [the struct that holds the External Name configuration]:
+
+```go
+// ExternalName contains all information that is necessary for naming operations,
+// such as removal of those fields from spec schema and calling Configure function
+// to fill attributes with information given in external name.
+type ExternalName struct {
+    // SetIdentifierArgumentFn sets the name of the resource in Terraform argument
+    // map.
+    SetIdentifierArgumentFn SetIdentifierArgumentFn
+    
+    // OmittedFields are the ones you'd like to be removed from the schema since
+    // they are specified via external name. You can omit only the top level fields.
+    // No field is omitted by default.
+    OmittedFields []string
+    
+    // DisableNameInitializer allows you to specify whether the name initializer
+    // that sets external name to metadata.name if none specified should be disabled.
+    // It needs to be disabled for resources whose external name includes information
+    // more than the actual name of the resource, like subscription ID or region
+    // etc. which is unlikely to be included in metadata.name
+    DisableNameInitializer bool
+}
+```
+
+Comments explain the purpose of each field but let's clarify further with some
+examples.
+
+Checking the [import section of aws_vpc], we see that this resource is being
```

Comment:
nitpick: I'd prefer to start with the usual case and continue with the `IdentifierFromProvider` case so that readers make the effort to do it properly and only if they fail, they continue with modifications.